File: src/SPA/src/app/_core/models/system/menuvm.ts
export interface MenuVM {
  id: string;
  name: string;
  url: string;
  sortOrder: number;
  parentId: string;
  icon: string;
  functionId: string;
  commandId: string;
  code: string;
  children?: MenuVM[];
  newLinkFlag: boolean;
  open?: boolean;
  selected?: boolean; // Check or not
  menuType: 'C' | 'F'; // c: menu, f button
}

File: src/ViewModels/System/MenuVM.cs
namespace ViewModels.System;

public class MenuVM
{
    public string Id { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string Url { get; set; } = string.Empty;

    public int SortOrder { get; set; }

    public string ParentId { get; set; } = string.Empty;

    public string Icon { get; set; } = string.Empty;
    public string FunctionId { get; set; } = string.Empty;
    public string CommandId { get; set; } = string.Empty;
    public string Code { get; set; } = string.Empty;
}
